src/share/classes/com/sun/javatest/regtest/exec/ProcessCommand.java

```diff
@@ -271,12 +300,14 @@ public void run() {
                 errCopier.join();
                 int exitCode = process.waitFor();
 
+                pre_alarm.cancel();
                 // if the timeout hasn't fired, cancel it as quickly as possible
                 alarm.cancel();
 
                 return getStatus(exitCode, statusScanner.exitStatus());
```

Comment:
I think you need to do something more here.   If the class killed child processes, you need to ensure that (probably) `Status.error` is returned with a suitable string. In other words, consider the case where you unlock a child process and the main process exits normally for some reason. That should *not* be grounds for the test to pass.

Reply:
I see the point, but I also had a case where after force quitting the child process, the parent test process is able to retrieve everything it needed from the child (which finished its task required for the parent before getting itself locked up) and be able to succeed just fine.

If we were to return error code in such cases, we would be generating failures in cases where the parent process was able to finish, but its children were not. Not necessarily always directly related to the case under test, more of a tangential failure, but I do see the value in reporting such cases. I will tweak it then.

Reply:
We now detect whether we had to cancel any child processes and return an error code if we do (this would fix any potential false positives, where the parent test process is free to proceed, but a child process is unable to throw any error status back at the parent, because the child itself is stuck)